src/Bicep.Core/Analyzers/Linter/Rules/NoUnusedVariablesRule.cs

```diff
@@ -35,24 +36,24 @@ override public IEnumerable<IDiagnostic> AnalyzeInternal(SemanticModel model)
 
             // variables must have a reference of type VariableAccessSyntax
             var unreferencedVariables = model.Root.Declarations.OfType<VariableSymbol>()
-                                    .Where(sym => !model.FindReferences(sym).OfType<VariableAccessSyntax>().Any());
-
+                .Where(sym => !model.FindReferences(sym).OfType<VariableAccessSyntax>().Any())
+                .Where(sym => sym.Name != MissingName);
```

Comment:
This relies on an implementation detail.  There should be a better way to check for this scenario.

Reply:
sym.NameSyntax.IsValid looks promising.

Reply:
Changed to your suggestion